src/main/java/org/opensearch/knn/index/engine/faiss/FaissIVFPQEncoder.java

```diff
@@ -90,4 +93,13 @@ public class FaissIVFPQEncoder implements Encoder {
     public MethodComponent getMethodComponent() {
         return METHOD_COMPONENT;
     }
+
+    @Override
+    public CompressionLevel calculateCompressionLevel(
+        MethodComponentContext methodComponentContext,
+        KNNMethodConfigContext knnMethodConfigContext
+    ) {
+        // TODO: For now, not supported out of the box
+        return CompressionLevel.NOT_CONFIGURED;
```

Comment:
same here, why is this not 1x? I thought not configured was when customer doensn't provide so we don't have to pass null, theoretically these cases are 1x then why not configured?

Reply:
Currently, the compression level returned here will be used to setup a default rescore context. So, it wont have a major impact. Im going to circle back in future to implement these, but want to try to merge today for release.